src/test/java/com/ksm/wstbackoffice/controller/AddressControllerTest.java

```diff
@@ -0,0 +1,111 @@
+package com.ksm.wstbackoffice.controller;
+
+import com.ksm.wstbackoffice.endpoint.EndPoint;
+import com.ksm.wstbackoffice.entity.Address;
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.filter.log.RequestLoggingFilter;
+import io.restassured.filter.log.ResponseLoggingFilter;
+import io.restassured.mapper.TypeRef;
+import io.restassured.specification.RequestSpecification;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeAll;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@SpringBootTest
+@ActiveProfiles("test")
+public class AddressControllerTest {
+
+    private static RequestSpecification spec;
+
+    @BeforeAll
+    public static void initSpec() {
+        spec = new RequestSpecBuilder()
+                .setContentType(JSON)
+                .setBaseUri(RestAssured.DEFAULT_URI)
+                .setPort(EndPoint.port)
+                .addFilter(new ResponseLoggingFilter())
+                .addFilter(new RequestLoggingFilter())
+                .build();
+    }
+
+    @Test
+    public void findByIdTest() {
+        Address address = given()
+                .spec(spec)
+                .pathParam("id", 1)
+                .when()
+                .get(EndPoint.addressesWithId)
+                .then()
+                .statusCode(200)
+                .extract()
+                .body()
+                .as(Address.class);
+
+        Assert.assertNotNull(address);
+        assertThat(address.getId(), Matchers.is(1L));
```

Comment:
Please try to do this without extracting the response.
End try to use extract only when you really need it

Reply:
done